beacon-chain/p2p/service.go

```diff
@@ -277,6 +277,14 @@ func (s *Service) Start() {
 // Stop the p2p service and terminate all peer connections.
 func (s *Service) Stop() error {
 	defer s.cancel()
+
+	for _, pid := range s.Peers().Active() {
+		_, err := s.Send(s.ctx, RPCTopicMappings[RPCGoodByeTopic], RPCGoodByeTopic, pid)
```

Comment:
This context is already cancelled so i would expect 100% of these to fail.

Reply:
for my understanding, at what layer of abstraction prior to this is the context cancelled?

Reply:
I am sorry, I gave incorrect feedback here. The context is not cancelled until this function returns (`defer s.cancel()`)